File: LeetCode/1431_Kids_With_the_Greatest_Number_of_Candies/solution.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    let max=0;
    for(let i=0;i<candies.length;i++){
        if(max<candies[i])max=candies[i]
    }
    let arr=[];
    for(let i=0;i<candies.length;i++){
        if((max-extraCandies)<=candies[i]) arr.push(true)
        else arr.push(false)
    }
    return arr;
};

File: LeetCode/1431_Kids_With_the_Greatest_Number_of_Candies/solution.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max = 0;
        for (int x : candies) {
            if (max < x) max = x;
        }
        vector<bool> arr;
        for (int x : candies) {
            if ((max - extraCandies) <= x) arr.push_back(true);
            else arr.push_back(false);
        }
        return arr;
    }
};